app/common/services/udsService.js

```diff
@@ -300,19 +300,37 @@ angular.module('RedhatAccess.common').factory('udsService', [
                             );
                             return deferred.promise;
                         },
-                        public: function (caseNumber, commentText) {
+                        public: function (caseNumber, commentText,hoursWorked) {
                             var deferred = $q.defer();
-                            uds.postPublicComments(
-                                function (response) {
-                                    deferred.resolve(response);
-                                },
-                                function (error) {
-                                    deferred.reject(error);
-                                },
-                                caseNumber,
-                                commentText
 
-                            );
+                            if(hoursWorked===undefined)
```

Comment:
Instead of checking this property in udsService.js check it in UDS.js, I think it will reduce lot of duplicate code in both these files. And we have done that for strata conditional calls also. So u would just pass this value to the UDS.js and there do the check.